htdocs/contact/card.php

```diff
@@ -374,28 +375,22 @@
 			$ret = $extrafields->setOptionalsFromPost($extralabels,$object);
 			if ($ret < 0) $error++;
 
-            $result = $object->update($contactid, $user);
+			$result = $object->update($contactid, $user);
 
 			if ($result > 0) {
 				// Categories association
-				// First we delete all categories association
```

Comment:
Why removing this ? If we sumbit form with 2 categories instead of 3, how is removed the 3st one ?

Reply:
Function setCategories in contact.class.php will do it, "add" or "del" when necessary

Reply:
@joseplluis @eldy 
yes you're right, the function "setCategories()" already deletes the categories that are no longer used